app/controllers/api/ratings_controller.rb

```diff
@@ -22,50 +32,39 @@ def rate_class
     def check_connection
         @connection = Connection.find_by(id: params[:id])
 
-        unless @connection
-            render_json_400("Connection with that id does not exist")
+        return render_json_400("Connection with that id does not exist") unless @connection
 
+        if @connection.in_progress?
```

Comment:
these conditions are a code smell. Usually they are good indicators that we could break all this logic into smaller components. Controllers should be as lean as possible. Use modules and services to handle code logic.

Reply:
I started doing something similar to that in before_actions, because I used to leave it al in the controller method. I just can't really tell what should and shouldn't be in the main method, because one time I ended up with just two lines of code and it seemed like the rest wat missing since it wasn't "visible"